### PR TITLE
Update Gradle to 8.5

### PR DESCRIPTION
**Pull request by reta**
### Description
Update Gradle to 8.5

### Issues Resolved
Part of https://github.com/opensearch-project/OpenSearch/issues/10334

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: test/shared/build.gradle [hunk 1]
```diff
@@ -46,24 +46,18 @@ configurations {
     // This will force the MR project to require jars from upstream project dependencies, which shouldn't be a problem since
     // there should only be one upstream project.
     compileClasspath {
-        beforeLocking {
```
Comment (reta) [inline test/shared/build.gradle:49]: See please https://github.com/gradle/gradle/pull/25126

**Review by Xtansia — APPROVED** [review]
Thanks for this @reta!

**Comment by reta** [comment]
> Thanks for this @reta!

Thanks @Xtansia , updated the OpenSearch version (that brings new Gradle plugins), have one issue to fix:

```
> Configure project :opensearch-spark-30                                              
The org.gradle.api.plugins.Convention type has been deprecated. This is scheduled to be removed in Gradle 9.0. Consult the upgrading guide for further information: https:
//docs.gradle.org/8.5/userguide/upgrading_version_8.html#deprecated_access_to_conventions                                                                                 
        at build_6nu7is0k4y2vnu5md65y4hre5$_run_closure1.doCall$original(/home/andriy.redko/Workspaces/OpenSearch-plugins/opensearch-hadoop/spark/sql-30/build.gradle:13)
        (Run with --stacktrace to get the full stack trace of this deprecation warning.)

```

**Comment by Xtansia** [comment]
@reta I can see one other error
```
* What went wrong:
Execution failed for task ':dist:generatePomFileForMainPublication'.

> Could not apply withXml() to generated POM
   > Could not get unknown property 'archivesName' for project ':dist' of type org.gradle.api.Project.
```

**Comment by reta** [comment]
> @reta I can see one other error

Yep, thanks @Xtansia , fixed it but didn't push till checks pass

**Comment by reta** [comment]
It seems like I may need to use 2.12.0-SNAPSHOT for OpenSearch build tools...

**Review by Xtansia — APPROVED** [review]

**Comment by Xtansia** [comment]
@reta Looks like you didn't end up needing to depend on a SNAPSHOT, so this should be good to merge right?

**Comment by reta** [comment]
> @reta Looks like you didn't end up needing to depend on a SNAPSHOT, so this should be good to merge right?

Correct, 2.11.1 did the trick, thanks @Xtansia !

**Pull request merged**